New datatable

pml984 opened the pull request.

pml984 committed:
Closes #21. Added Ag-Grid with Material Theme

jeffshaver commented:
I am having issues. I ran `npm install` at both the top level and the examples directory.

I get errors trying to go into `BasicTable` (new?) about getting a null element in React.createElement.
When i go to `Table`, I get a table, that has no data in it (has rows, just no text/data).

It is also very slow to switch between elements.

cafedavid commented:
i ran `npm install` at top level and received several errors. in the examples directory, however, it worked.

`BasicTable` appeared to work correctly

Also noticed slowness when navigating between components

jeffshaver commented on examples/html/index.html:
```diff
@@ -6,15 +7,25 @@
   <link rel="stylesheet" href="flexboxgrid/dist/flexboxgrid.min.css">
   <link rel="stylesheet" href="leaflet/dist/leaflet.css">
+  <link rel="stylesheet" href="ag-grid/dist/styles/ag-grid.css">
+  <!--<link rel="stylesheet" href="ag-grid/dist/styles/theme-fresh.css">-->
```
this should be removed

jeffshaver commented on examples/html/index.html:
```diff
@@ -6,15 +7,25 @@
+  <link rel="stylesheet" href="safe-framework/dist/theme-material.css">
   <style>
-    .leaflet-container {
+  .leaflet-container {
```
shouldn't have been un-indented

pml984 commented:
I fixed the few issues you mentioned. I'm not sure why there are any errors occurring when running `npm install`. I am not seeing them when building off of this branch. I also created a new vm using this branch and did not see any errors when building.

For the performance issue, I did notice that the bundle.js file is a large file, 3mb+ prior to me adding these changes. It did increase that file size some when I added in the ag-grid stuff, but not enough to really increase the load time that much. It already took several seconds to load before I added that. I can look at ways to reduce the size of that file in both dev and prod builds.

When switching between components on the left bar, all load very fast as they did prior to me adding in the new data table.

jeffshaver commented:
@pml984 so its working now. everything is loading fast again when switching routes (idk what was going on last time). intiail load is fine. we will worry about bundle.js later. We aren't minifying right now (unless NODE_ENV=production).

However, I would like to try to style the tables slightly more like the BasicTable. The interesting part of this will be that we need access to the MUI theme in order to style it correctly.

Would like to have:
- alternating row colors
- lighter header color
- auto-width columns (so that the table takes up 100% of the width as a minimum and then does horizontal scrolling when it needs to)

Minus the slight styling changes, I think this is great!

jeffshaver commented:
Also, maybe some cell padding? The checkbox on the left side seems awfully close to the edge

pml984 commented:
Made all the changes. There may still be some smaller changes that need to be made, but most of the major issues should be resolved.
